src/StackExchange.Redis/Interfaces/IDatabase.cs

```diff
@@ -552,11 +553,14 @@ public interface IDatabase : IRedis, IDatabaseAsync
         /// Since Redis 2.1.3, you can update the timeout of a key.
         /// It is also possible to remove the timeout using the PERSIST command. See the page on key expiry for more information.
         /// </para>
+        /// <para>
+        /// Since Redis 7.0.0, you can choose on which condition the timeout will be set using <see cref="ExpiryOption"/>. Notice that this option cannot be used with PERSIST command.
+        /// </para>
         /// </remarks>
         /// <remarks>https://redis.io/commands/expire</remarks>
         /// <remarks>https://redis.io/commands/pexpire</remarks>
         /// <remarks>https://redis.io/commands/persist</remarks>
-        bool KeyExpire(RedisKey key, TimeSpan? expiry, CommandFlags flags = CommandFlags.None);
+        bool KeyExpire(RedisKey key, TimeSpan? expiry, CommandFlags flags = CommandFlags.None, ExpiryOption? expiryOption = null);
```

Comment:
This isn't a change we can make - when adding a parameter like this it'll break all current consumers. We'll need to add a new overload and think about how this API looks. Think of it this way: any consumer has a binary `.KeyExpire(RedisKey, TimeSpan?, CommandFlags)` method it's looking for - if we do this, it removes that method and they get a `MissingMethodException` at runtime.

Reply:
Then I can add 
`bool KeyExpire(RedisKey key, TimeSpan? expiry, ExpiryOption expiryOption, CommandFlags flags = CommandFlags.None);`
?

Reply:
@NickCraver

Reply:
If it helps, you can take a look at Jedis' PR (the Java client): https://github.com/redis/jedis/pull/2852/files

Reply:
Gotcha - I'll try and  grok it a little further along (have some other TODOs this AM) - may suggest naming tweaks to clarify!

Reply:
Sounds great! Apologies I'm a bit slow - y'all are rocking it, hard to keep up with :)